pkg/plugin/manager.go

```diff
@@ -0,0 +1,196 @@
+					// We save it so that we do not have to do this parsing again.
+					m.persist(plugin)
+				}
+			}
+
+			plugins = append(plugins, plugin)
+		}
+	}
+
+	return plugins
+}
+
+// loadPluginFromPackage fills a Plugin structure from a Universe package.
+// This is based on the assumption that old packages were all downloaded
+// using the Universe and are thus Universe packages.
+func (m *Manager) loadPluginFromPackage(plugin *Plugin) error {
+	packageFilePath := filepath.Join(plugin.dir, "package.json")
+	data, err := ioutil.ReadFile(packageFilePath)
+	if err != nil {
+		return err
+	}
+
+	pkg := &Package{}
+
+	if err = json.Unmarshal(data, pkg); err != nil {
+		return err
+	}
+
+	// We transfer the information we have from the package to the plugin.
+	plugin.Name = pkg.Name
+	plugin.Description = pkg.Description
+	plugin.Version = pkg.Version
+
+	// Create a handle to then get all the binaries.
+	binDirHandle, err := m.Fs.Open(plugin.binDir)
+	if err != nil {
+		return err
+	}
+	defer binDirHandle.Close()
+
+	// Read the directory to get all the binaries.
+	binaries, err := binDirHandle.Readdir(0)
+	if err != nil {
+		return err
+	}
+
+	executables := []*executable{}
+
+	// Loop over the binaries to create the list of plugin executables.
+	for _, binary := range binaries {
+		if strings.HasPrefix(binary.Name(), "dcos-") {
+			commandName := strings.TrimPrefix(binary.Name(), "dcos-")
+			if runtime.GOOS == "windows" {
+				commandName = strings.TrimSuffix(commandName, ".exe")
+			}
+
+			cmd := &command{
+				Name: commandName,
+			}
+
+			// We add the command information to the plugin file by shelling it out.
+			cmdExe := filepath.Join(plugin.binDir, binary.Name())
+			infoCmd, err := exec.Command(cmdExe, commandName, "--info").Output()
+			if err != nil {
+				log.Fatal(err)
+			}
+			cmd.Description = strings.TrimSpace(string(infoCmd))
+
+			// We create an executable which is what will be executed when calling a subcommand.
+			exe := &executable{
+				Filename: binary.Name(),
+				Commands: []*command{
+					cmd,
+				},
+			}
+			executables = append(executables, exe)
+		}
+	}
+
+	plugin.Executables = executables
+	return nil
+}
+
+// Persist saves a `plugin.yaml` file representing a plugin if it does not exist.
+func (m *Manager) persist(plugin *Plugin) error {
+	// We need an env directory for `plugin.yaml`.
+	envFilePath := filepath.Join(plugin.dir, "env")
+	if _, err := os.Stat(envFilePath); os.IsNotExist(err) {
+		return errors.New(envFilePath + " does not exist")
+	}
+
+	// We should not overwrite `plugin.yaml`, this method should only be used
+	// to create a minimal `plugin.yaml` if the user has an old plugin.
+	pluginFilePath := filepath.Join(plugin.dir, "env", "plugin.yaml")
+	if _, err := os.Stat(pluginFilePath); err == nil {
+		return errors.New(pluginFilePath + " already exists")
+	}
+
+	// Create the `plugin.yaml` file.
+	pluginFile, err := m.Fs.Create(pluginFilePath)
```

Comment:
The file is not closed. Instead of creating / writing to the file, you could use `afero.WriteFile(m.Fs, pluginFilePath, pluginYAML, 0644)`